istioctl/pkg/multicluster/remote_secret.go

```diff
@@ -211,14 +214,10 @@ var (
 	errMissingTokenKey  = fmt.Errorf("no %q data found", v1.ServiceAccountTokenKey)
 )
 
-func createRemoteSecretFromTokenAndServer(tokenSecret *v1.Secret, clusterName, server, secName string) (*v1.Secret, error) {
-	caData, ok := tokenSecret.Data[v1.ServiceAccountRootCAKey]
-	if !ok {
-		return nil, errMissingRootCAKey
-	}
-	token, ok := tokenSecret.Data[v1.ServiceAccountTokenKey]
-	if !ok {
-		return nil, errMissingTokenKey
+func createRemoteSecretFromTokenAndServer(client kube.ExtendedClient, tokenSecret *v1.Secret, clusterName, server, secName string) (*v1.Secret, error) {
+	caData, token, err := waitForTokenData(client, tokenSecret)
```

Comment:
Should `kube client` be checked here for nil value rather than only check it in `waitForTokenData` at L235?

Reply:
That check was just to make tests pass.. without the "wait" logic.

Reply:
I just got rid of that check and added proper tests.

Reply:
oh, got it, thanks @stevenctl